### PR TITLE
[DOCS] Explain flood stage watermark.

### LINKED ISSUE #36114: Delete documents via delete_by_query if index.blocks.read_only_allow_delete is set to true
**Elasticsearch version** (`bin/elasticsearch --version`): 6.5.1

**Plugins installed**: []

**JVM version** (`java -version`): 1.8.0u191

**OS version** (`uname -a` if on a Unix-like system): 4.4.0-128-generic

**Description of the problem including expected versus actual behavior**:

Elasticsearch does not allow to delete documents via delete_by_query api if `index.blocks.read_only_allow_delete` is set to `true`.
The error is
```
{"took":6,"timed_out":false,"total":1,"deleted":0,"batches":1,"version_conflicts":0,"noops":0,"retries":{"bulk":0,"search":0},"throttled_millis":0,"requests_per_second":-1.0,"throttled_until_millis":0,"failures":[{"index":"my-index","type":"doc","id":"1234","cause":{"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"},"status":403}]}
```

I would expect that deletion via the delete_by_query api is allowed if `index.blocks.read_only_allow_delete` is set to `true`.

**Steps to reproduce**:

 1. Create an index 
 2. Insert data
 3. Set `index.blocks.read_only_allow_delete` to `true`
 4. Try to delete data using the `delete_by_query` api

**Provide logs (if relevant)**:

### PR DESCRIPTION
**Pull request by lockewritesdocs**
This PR introduces changes for #36114, which include:
* Explaining the difference between deleting documents from an index and deleting the index itself.
* Clarifications to the `index.blocks.read_only_allow_delete` parameter.
* Adding a note that ES adds and removes the read-only index block on its own.

See the [Dynamic index settings](https://elasticsearch_57184.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/index-modules.html#dynamic-index-settings) page on the docs preview server for changes.

Closes #36114 

Backports:
* [7.8] #57301
* [7.7] #57318
* [7.6] #57320
* [7.x] #57321
* [6.8] #57322

### CODE REVIEW

**Comment by lockewritesdocs** [comment]
@elasticmachine

File: docs/reference/modules/cluster/disk_allocator.asciidoc [hunk 1]
```diff
@@ -37,23 +37,23 @@ file or updated dynamically on a live cluster with the
...
-values. This is so that we can we validate that the settings are internally
-consistent (that is, the low disk threshold is not more than the high disk
-threshold, and the high disk threshold is not more than the flood stage
-threshold).
+NOTE: You cannot mix the usage of percentage values and byte values within
+these settings. Either all values are set to percentage values, or all are set to byte
+values. This enforcement is so that Elasticsearch can validate that the settings are internally
+consistent, ensuring that the low disk threshold is less than the high disk
+threshold, and the high disk threshold is less than the flood stage
+threshold. Therefore, you typically do not need to modify this setting.
```
Comment (nik9000) [inline docs/reference/modules/cluster/disk_allocator.asciidoc:56]: I don't think the last sentence follows from the description. I think it'd be lovely to have a paragraph on when it is useful to modify these settings, but that it'd belong more at the top of the page than here. I think if you modify the settings you typically modify all of them. I *think* you mostly are interested in modifying them if you have a very big disk and 85%/90%/95% would be quite wasteful. Of if you write *lots* of data very quickly compared to the amount of disk so the defaults aren't conservative enough.

Comment (lockewritesdocs) [reply docs/reference/modules/cluster/disk_allocator.asciidoc]: I removed the last sentence [this commit](https://github.com/elastic/elasticsearch/pull/57184/commits/f82d51a552c8988ad4f4053ca3ed0914845bf126). If you think an expanded explanation of why users might want to modify these settings is warranted, let me know.

Comment (nik9000) [reply docs/reference/modules/cluster/disk_allocator.asciidoc]: I do think it'd be nice, but this is a good change as it stands now anyway. Merge anytime you are ready!

**Review by nik9000 — APPROVED** [review]

**Pull request merged**